packages/desktop-client/src/components/accounts/Balance.tsx

```diff
@@ -42,32 +56,37 @@ function DetailedBalance({ name, balance, isExactBalance = true }) {
   );
 }
 
-function SelectedBalance({ selectedItems, account }) {
+type SelectedBalanceProps = {
+  selectedItems: Set<string>;
+  account: AccountEntity;
+};
+
+function SelectedBalance({ selectedItems, account }: SelectedBalanceProps) {
   const { t } = useTranslation();
 
   const name = `selected-balance-${[...selectedItems].join('-')}`;
 
-  const rows = useSheetValue({
-    name,
+  const rows = useSheetValue<'balance', `selected-balance-${string}`>({
+    name: name as `selected-balance-${string}`,
     query: q('transactions')
       .filter({
         id: { $oneof: [...selectedItems] },
         parent_id: { $oneof: [...selectedItems] },
       })
       .select('id'),
   });
-  const ids = new Set((rows || []).map(r => r.id));
+  const ids = new Set(Array.isArray(rows) ? rows.map(r => r.id) : []);
 
   const finalIds = [...selectedItems].filter(id => !ids.has(id));
-  let balance = useSheetValue({
-    name: name + '-sum',
+  let balance = useSheetValue<'balance', `selected-balance-${string}`>({
+    name: (name + '-sum') as `selected-balance-${string}`,
     query: q('transactions')
       .filter({ id: { $oneof: finalIds } })
       .options({ splits: 'all' })
       .calculate({ $sum: '$amount' }),
   });
 
-  let scheduleBalance = null;
+  let scheduleBalance = 0;
```

Comment:
While changing ```scheduleBalance``` from ```null``` to ```0``` fixes type checking, line 121 checks to see if ```scheduleBalance``` is ```null```.  On line 121, ```scheduleBalance``` will never be ```null```.
If ```balance``` is never ```null``` then you could streamline the if statements on lines 121-125.

Reply:
good catch! updated.